q2_moshpit/busco/types/_transformer.py

```diff
@@ -0,0 +1,40 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2022-2023, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+import pandas as pd
+from qiime2 import Metadata
+
+from . import BUSCOResultsFormat
+from ...plugin_setup import plugin
+
+
+def _read_dataframe(fh: str, header=0):
+    df = pd.read_csv(fh, sep='\t', header=header, index_col=0, dtype='str')
+    df.index.name = 'id'
+    return df
+
+
+@plugin.register_transformer
+def _1(ff: BUSCOResultsFormat) -> pd.DataFrame:
+    with ff.open() as fh:
+        df = _read_dataframe(fh)
+        return df
+
+
+@plugin.register_transformer
+def _2(data: pd.DataFrame) -> BUSCOResultsFormat:
+    ff = BUSCOResultsFormat()
+    with ff.open() as fh:
+        data.to_csv(fh, sep='\t', index=False, header=True)
+    return ff
+
+
+@plugin.register_transformer
+def _3(ff: BUSCOResultsFormat) -> Metadata:
+    with ff.open() as fh:
+        df = _read_dataframe(fh)
+        return Metadata(df)
```

Comment:
Transformer nubie here! Just curious, is it necessary to call `.open()` on this `BUSCOResultsFormat` or would it suffice to do something like this?

```suggestion
@plugin.register_transformer
def _1(ff: BUSCOResultsFormat) -> pd.DataFrame:
     df = _read_dataframe(ff.path)
     return df


@plugin.register_transformer
def _2(data: pd.DataFrame) -> BUSCOResultsFormat:
    ff = BUSCOResultsFormat()
    data.to_csv(ff.path, sep='\t', index=False, header=True)
    return ff


@plugin.register_transformer
def _3(ff: BUSCOResultsFormat) -> Metadata:
     df = _read_dataframe(ff.path)
     return Metadata(df)
```

Reply:
Hmmm, I don't think I tried that so I'm not 100% sure, but my guess would be probably yes? Wanna try it out and let me know? 😆

Reply:
> Wanna try it out and let me know? 😆

It works! I made the changes locally and all the tests passed ✅. I also tried the action with a small input and I could not find anything wrong with the visualization.